src/client/pythonEnvironments/discovery/locators/services/pipEnvHelper.ts

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as path from 'path';
+import { traceError } from '../../../../common/logger';
+import { getEnvironmentVariable } from '../../../../common/utils/platform';
+import { arePathsSame, pathExists, readFile } from '../../../common/externalDependencies';
+
+function getSearchDepth() {
+    // PIPENV_MAX_DEPTH tells pipenv the maximum number of directories to recursively search for
+    // a Pipfile, defaults to 3: https://pipenv.pypa.io/en/latest/advanced/#pipenv.environments.PIPENV_MAX_DEPTH
+    const maxDepthStr = getEnvironmentVariable('PIPENV_MAX_DEPTH');
+    if (maxDepthStr !== undefined) {
+        const maxDepth = parseInt(maxDepthStr, 10);
+        // eslint-disable-next-line no-restricted-globals
+        if (isNaN(maxDepth)) {
+            traceError(`PIPENV_MAX_DEPTH is incorrectly set. Converting value '${maxDepthStr}' to number results in NaN`);
+            return 1;
+        }
+        return maxDepth;
+    }
+    return 3;
+}
+
+/**
+ * Returns the path to Pipfile associated with the provided directory.
+ * @param searchDir the directory to look into
+ * @param lookIntoParentDirectories set to true if we should also search for Pipfile in parent directory
+ */
+export async function _getAssociatedPipfile(
+    searchDir: string,
+    options: {lookIntoParentDirectories: boolean},

```

Comment:
IIRC, you don't need "options:".

```suggestion
    { lookIntoParentDirectories: boolean },
```

Reply:
Huh. Actually I'm not sure how to access such a parameter. Also ESlint expects to strongly type this.
`{ lookIntoParentDirectories: boolean }: <non-any type>`

Reply:
If this is what you're referring to,
```
export async function _getAssociatedPipfile(
    searchDir: string,
    { lookIntoParentDirectories }: { lookIntoParentDirectories: boolean },
): Promise<string | undefined> {
    const pipFileName = getEnvironmentVariable('PIPENV_PIPFILE') || 'Pipfile';
    let heightToSearch = { lookIntoParentDirectories }.lookIntoParentDirectories ? getSearchHeight() : 1;
    ...
}
```
It looks weird.